packages/token/src/erc721/extensions/erc721_wrapper.cairo

```diff
@@ -0,0 +1,171 @@
+            let mut erc721_component = get_dep_component_mut!(ref self, ERC721);
+            let data = array![].span();
+
+            for token_id in token_ids {
+                // This is an "unsafe" transfer that doesn't call any hook on the receiver. With
+                // underlying() being trusted (by design of this contract) and no other contracts
+                // expected to be called from there, reentrancy should be safe.
+                token.transfer_from(caller, this, *token_id);
+                erc721_component.safe_mint(receiver, *token_id, data);
+            }
+            true
+        }
+
+        /// Burns wrapped tokens from the caller and withdraws underlying tokens to `receiver`.
+        fn withdraw_to(
+            ref self: ComponentState<TContractState>,
+            receiver: ContractAddress,
+            token_ids: Span<u256>,
+        ) -> bool {
+            let caller = starknet::get_caller_address();
+            let this = starknet::get_contract_address();
+            let underlying = self.underlying();
+            let token = IERC721Dispatcher { contract_address: underlying };
+            let mut erc721_component = get_dep_component_mut!(ref self, ERC721);
+            let data = array![].span();
+
+            for token_id in token_ids {
+                // Setting an "auth" arguments enables the `_check_authorized` check which verifies
+                // that the token exists (owner != 0). Therefore, it is not needed to verify that
+                // the return value is not 0 here.
+                erc721_component.update(Zero::zero(), *token_id, caller);
+                // Checks were already performed at this point, and there's no way to retake
+                // ownership or approval from the wrapped token id after this point.
+                token.safe_transfer_from(this, receiver, *token_id, data);
+            }
+            true
+        }
+    }
+
+    #[embeddable_as(ERC721WrapperReceiverImpl)]
+    impl ERC721WrapperReceiver<
+        TContractState,
+        +HasComponent<TContractState>,
+        impl ERC721: ERC721Component::HasComponent<TContractState>,
+        +ERC721Component::ERC721HooksTrait<TContractState>,
+        +SRC5Component::HasComponent<TContractState>,
+        +Drop<TContractState>,
+    > of IERC721ReceiverMut<ComponentState<TContractState>> {
+        /// Accepts safe transfers of the underlying token and mints the wrapped token.
+        fn on_erc721_received(
+            ref self: ComponentState<TContractState>,
+            operator: ContractAddress,
+            from: ContractAddress,
+            token_id: u256,
+            data: Span<felt252>,
+        ) -> felt252 {
+            let caller = starknet::get_caller_address();
+            assert(caller == self.underlying(), Errors::UNSUPPORTED_TOKEN);
+
+            let mut erc721_component = get_dep_component_mut!(ref self, ERC721);
+            // TODO!: Should mint to from or operator?
```

Comment:
It should mint to `from`, based on [Solidity reference](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/extensions/ERC721Wrapper.sol#L79).